voluptuous/humanize.py

```diff
@@ -0,0 +1,39 @@
+from voluptuous import Invalid, MultipleInvalid
+from voluptuous.error import Error
+
+
+MAX_VALIDATION_ERROR_ITEM_LENGTH = 500
+
+
+def _nested_getitem(data, path):
+    for item_name in path:
+        try:
```

Comment:
Ok. Let's avoid `reduce` due to python3 compatibility issue. However it would be good if you can use conditionals instead of exceptionals.